src/main/java/org/wso2/carbon/apimgt/securityenforcer/internal/PingAIHandlerComponent.java

```diff
@@ -0,0 +1,199 @@
+        } catch (IOException | ParseException e) {
+            log.error("Error when reading the payload", e);
+        }
+        return managementAPIPayloadJson;
+    }
+
+    private void logConfigData(AISecurityHandlerConfig securityHandlerConfig) {
+
+        if (log.isDebugEnabled()) {
+            try {
+                String logMessage = "Ping AI configurations- ";
+                logMessage = logMessage + ", Operation Mode: " + securityHandlerConfig.getMode();
+                logMessage = logMessage + ", Cache Expiry time: " + securityHandlerConfig.getCacheExpiryTime();
+                logMessage = logMessage + ", ASE Endpoint: " + securityHandlerConfig.getAseConfig().getEndPoint();
+                logMessage =
+                        logMessage + ", Management Endpoint: " + securityHandlerConfig.getModelCreationEndpointConfig()
+                                .getManagementAPIEndpoint();
+                logMessage =
+                        logMessage + ", DataPublisher- MaxPerRoute: " + securityHandlerConfig.getDataPublisherConfig()
+                                .getMaxPerRoute();
+                logMessage = logMessage + ", DataPublisher- MaxOpenConnections: " + securityHandlerConfig
+                        .getDataPublisherConfig().getMaxOpenConnections();
+                logMessage = logMessage + ", DataPublisher- ConnectionTimeout: " + securityHandlerConfig
+                        .getDataPublisherConfig().getConnectionTimeout();
+                logMessage = logMessage + ", ThreadPoolExecutor- CorePoolSize: " + securityHandlerConfig
+                        .getThreadPoolExecutorConfig().getCorePoolSize();
+                logMessage = logMessage + ", ThreadPoolExecutor- MaximumPoolSize: " + securityHandlerConfig
+                        .getThreadPoolExecutorConfig().getMaximumPoolSize();
+                logMessage = logMessage + ", ThreadPoolExecutor- KeepAliveTime: " + securityHandlerConfig
+                        .getThreadPoolExecutorConfig().getKeepAliveTime();
+                logMessage =
+                        logMessage + ", StackObjectPool- MaxIdle: " + securityHandlerConfig.getStackObjectPoolConfig()
+                                .getMaxIdle();
+                logMessage = logMessage + ", StackObjectPool- InitIdleCapacity: " + securityHandlerConfig
+                        .getStackObjectPoolConfig().getInitIdleCapacity();
+                if (securityHandlerConfig.getLimitTransportHeaders().isEnable()) {
+                    logMessage =
+                            logMessage + ", LimitTransportHeaders: " + securityHandlerConfig.getLimitTransportHeaders()
+                                    .getHeaderSet().toString();
+                } else {
+                    logMessage = logMessage + ", Limit Transport headers Disabled";
+                }
+                log.debug(logMessage);
+            } catch (Exception e) {
```

Comment:
Any reason are we catching a generic exception here? I noticed this in some other places as well.

Reply:
fixed with https://github.com/wso2-extensions/apim-handler-pingai/pull/22